CI.rst

```diff
@@ -629,6 +691,90 @@ delete old artifacts that are > 7 days old. It only runs for the 'apache/airflow
 We also have a script that can help to clean-up the old artifacts:
 `remove_artifacts.sh <dev/remove_artifacts.sh>`_
 
+Selective CI Checks
+===================
+
+In order to optimise our CI builds, we've implemented optimisations to only run selected checks for some
+kind of changes. The logic implemented reflects the internal architecture of Airflow 2.0 packages
+and it helps to keep down both the usage of jobs in GitHub Actions as well as CI feedback time to
+contributors in case of simpler changes.
+
+We have the following test types (separated by packages in which they are):
+
+* Core - for the core Airflow functionality (core folder)
```

Comment:
There isn't a core folder -- do you mean airflow/, if not one of the more specific examples below?

Reply:
I actually moved the tests to "core" folder in the previous PR because I needed cleaner "per directory" split of tests in pytest. In pytest it is far easier to say "run tests" from that directory rather than "run tests from all the files in this directory but do not recurse to this specific directory". So I moved all the tests that were in "airflow/*.py" to "airflow/core/"